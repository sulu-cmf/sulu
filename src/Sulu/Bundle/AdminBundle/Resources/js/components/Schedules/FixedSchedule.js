// @flow
import React from 'react';
import {translate} from '../../utils/Translator';
import DatePicker from '../DatePicker';
import Form from '../Form';
import type {FixedScheduleEntry} from './types';

type Props = {|
    index: number,
    onChange: (index: number, value: FixedScheduleEntry) => void,
    value: FixedScheduleEntry,
|};

export default class FixedSchedule extends React.Component<Props> {
    handleStartChange = (start: ?Date) => {
        const {index, onChange, value} = this.props;
        onChange(index, {...value, start});
    };

    handleEndChange = (end: ?Date) => {
        const {index, onChange, value} = this.props;
        onChange(index, {...value, end});
    };

    render() {
        const {value} = this.props;

        return (
            <Form>
                <Form.Field colSpan={6} label={translate('sulu_admin.start')}>
                    <DatePicker
                        onChange={this.handleStartChange}
                        options={{dateFormat: true, timeFormat: true}}
                        value={value.start}
                    />
                </Form.Field>
                <Form.Field colSpan={6} label={translate('sulu_admin.end')}>
                    <DatePicker
                        onChange={this.handleEndChange}
                        options={{dateFormat: true, timeFormat: true}}
                        value={value.end}
                    />
                </Form.Field>
            </Form>
        );
    }
}
