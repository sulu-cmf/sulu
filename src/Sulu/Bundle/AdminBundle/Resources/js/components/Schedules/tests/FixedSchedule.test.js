// @flow
import React from 'react';
import {render, shallow} from 'enzyme';
import FixedSchedule from '../FixedSchedule';

jest.mock('../../../utils/Translator', () => ({
    translate: jest.fn((key) => key),
}));

test('Render a FixedSchedule', () => {
    expect(render(<FixedSchedule index={0} onChange={jest.fn()} value={{type: 'fixed'}} />)).toMatchSnapshot();
});

test('Change start date', () => {
    const date = new Date();

    const changeSpy = jest.fn();
    const fixedSchedule = shallow(
        <FixedSchedule index={0} onChange={changeSpy} value={{end: date, type: 'fixed'}} />
    );

    fixedSchedule.find('Field[label="sulu_admin.start"] DatePicker').simulate('change', date);
    expect(changeSpy).toBeCalledWith(0, {end: date, start: date, type: 'fixed'});
});

test('Change end date', () => {
    const date = new Date();

    const changeSpy = jest.fn();
    const fixedSchedule = shallow(
        <FixedSchedule index={2} onChange={changeSpy} value={{start: date, type: 'fixed'}} />
    );

    fixedSchedule.find('Field[label="sulu_admin.end"] DatePicker').simulate('change', date);
    expect(changeSpy).toBeCalledWith(2, {end: date, start: date, type: 'fixed'});
});
