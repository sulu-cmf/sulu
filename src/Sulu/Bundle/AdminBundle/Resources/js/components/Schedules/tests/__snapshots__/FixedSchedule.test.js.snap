// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render a FixedSchedule 1`] = `
<div
  class="grid grid"
>
  <div
    class="item gridItem colSpan colSpan-6 space-before-0 space-after-0"
  >
    <div
      class="field"
    >
      <label
        class="label"
      >
        sulu_admin.start
      </label>
      <div>
        <div />
      </div>
      <label
        class="errorLabel"
      />
    </div>
  </div>
  <div
    class="item gridItem colSpan colSpan-6 space-before-0 space-after-0"
  >
    <div
      class="field"
    >
      <label
        class="label"
      >
        sulu_admin.end
      </label>
      <div>
        <div />
      </div>
      <label
        class="errorLabel"
      />
    </div>
  </div>
</div>
`;
